### PR TITLE
go test -coverpkg=./...

### PR DESCRIPTION
**Pull request by nicksnyder**
https://github.com/sourcegraph/sourcegraph/issues/10356

### CODE REVIEW

**Comment by unknwon** [comment]
Dropping random observation: Not verified, but this one looks like somehow the assumption of "working directory" was broken when `-coverpkg` flag is supplied:

```
2020/05/06 04:24:48 Failed to find templates directory: go [-e -json -compiled=false -test=false -export=false -deps=false -find=true -- github.com/sourcegraph/sourcegraph/cmd/frontend/internal/app/templates]: exit status 1: failed to initialize build cache at /dev/null/.cache/go-build: mkdir /dev/null: not a directory
```

**Comment by keegancsmith** [comment]
Came across this https://www.ory.sh/golang-go-code-coverage-accurate/

Not using the tool yet, but rather just starting off with the shell script they mention in the README.

**Comment by unknwon** [comment]
Seems failed on shellcheck? 🤦 https://buildkite.com/sourcegraph/sourcegraph/builds/62798#0d1feb21-2cfc-4a98-9281-fd2e6493e212/330-334

**Comment by unknwon** [comment]
For some reason, it lowered our overall coverage... For comparison:
- Lastest commit on `master`, 44.25%: https://codecov.io/gh/sourcegraph/sourcegraph/tree/5b57d276686009f6344f028e5dfe87876e40667b
- This PR, 33.2%: https://codecov.io/gh/sourcegraph/sourcegraph/tree/7401e56ceff697bda7eae3c9549bbd9f7b9e7e93

**Comment by keegancsmith** [comment]
> For some reason, it lowered our overall coverage...

See my comment above. The shell script is suboptimal, it excludes
every package containing cmd/. But its a good sign that it ran, we can
improve it from here.

File: dev/ci/go-test.sh [hunk 1]
```diff
@@ -14,5 +14,11 @@ echo "--- comby install"
 echo "--- go mod download"
 go mod download
 
+echo "--- go coverage"
+# From https://github.com/ory/go-acc
+touch ./coverage.tmp
+echo 'mode: atomic' > coverage.txt
+go list ./... | grep -v /cmd | grep -v /vendor | xargs -n1 -I{} sh -c 'go test -race -covermode=atomic -coverprofile=coverage.tmp -coverpkg $(go list ./... | grep -v /vendor | tr "\n" ",") {} && tail -n +2 coverage.tmp >> coverage.txt || exit 255' && rm coverage.tmp
```
Comment (nicksnyder) [inline dev/ci/go-test.sh:21]:
```suggestion
go list ./... | grep -v /vendor | xargs -n1 -I{} sh -c 'go test -race -covermode=atomic -coverprofile=coverage.tmp -coverpkg $(go list ./... | grep -v /vendor | tr "\n" ",") {} && tail -n +2 coverage.tmp >> coverage.txt || exit 255' && rm coverage.tmp
```

**Comment by unknwon** [comment]
Fixed shellcheck in #11332 and passed CI, but it took over 16m for `go-test.sh` step (which [usually took <2m](https://buildkite.com/sourcegraph/sourcegraph/builds/65780)).

The coverage increase +3.17% for "go".

**Comment by nicksnyder** [comment]
Going to close this PR since it won’t be merged. It seems like our currently measured coverage is pretty accurate so maybe the original issue should be closed and we should simply write more tests :)

**Pull request closed without merging**